packages/neon_framework/packages/neon_http_client/lib/src/interceptors/authorization_throttling_interceptor.dart

```diff
@@ -49,15 +52,52 @@ final class AuthorizationThrottlingInterceptor implements HttpInterceptor {
   }
 
   @override
-  StreamedResponse interceptResponse({required StreamedResponse response, required Uri url}) {
+  Future<StreamedResponse> interceptResponse({required StreamedResponse response, required Uri url}) async {
     assert(
       shouldInterceptResponse(response),
       'Response should not be intercepted.',
     );
 
     final authorization = response.request!.headers['authorization'];
     if (authorization != null && authorization.isNotEmpty && response.statusCode == 401) {
-      _blocked = true;
+      final contentType = response.headers['content-type'];
+      if (contentType == null) {
+        return response;
+      }
+
+      final mediaType = MediaType.parse(contentType);
```

Comment:
IIrc this might throw a FormatException.
Please check and catch the appropriate one.